getsentry/sentry-cocoa
# Allow compiling with APPLICATION_EXTENSION_API_ONLY=NO

## :scroll: Description


⚠️ **Exposes internal methods** to allow compiling with `APPLICATION_EXTENSION_API_ONLY=NO`.

I was expecting that setting `APPLICATION_EXTENSION_API_ONLY=NO` would cause no issues since [it is supposed to be less restricting allowing the usage of forbidden extension API](https://github.com/getsentry/sentry-react-native/issues/3908#issuecomment-2438136419). 
In reality as soon as you enable `Allow app extension API only` the generated header will contain both public and internal classes ([reference](https://stackoverflow.com/questions/42319892/checking-allow-app-extension-api-only-in-ios-framework-project-exposes-interna), [reference 2](https://forums.developer.apple.com/forums/thread/72768)). This means that disabling the flag hides many of the internal declarations that the project needs to compile.

This PR exposes those classes allowing for the project to compile with `APPLICATION_EXTENSION_API_ONLY` both set to `YES` and `NO`.

## :bulb: Motivation and Context



Fixes https://github.com/getsentry/sentry-cocoa/issues/4426 and https://github.com/getsentry/sentry-react-native/issues/3908

## :green_heart: How did you test it?
1. Compile and run `iOS-Swift` project on the `main` branch
2. Select the `Sentry` target and set `APPLICATION_EXTENSION_API_ONLY` to `NO`
3. Notice that compilation fails with random errors ❌ 
4. Switch to this branch
5. Compile and run `iOS-Swift` project with `APPLICATION_EXTENSION_API_ONLY` both set to `YES` and `NO` ✅ 

## :pencil: Checklist

You have to check all boxes before merging:

- [ ] Review from the native team if needed.

## :crystal_ball: Next steps

If this is is a viable solution:
- fix failing tests
- clean up the code
- document the changes
- add CI checks compiling the project with `APPLICATION_EXTENSION_API_ONLY` both set to `YES` and `NO`
